src/frontend/contexts/products-context.js

```diff
@@ -0,0 +1,30 @@
+import { useState, useEffect, createContext, useContext } from "react";
+const { default: axios } = require("axios");
+
+const ProductsContext = createContext();
+
+const ProductsProvider = ({ children }) => {
+    const [products, setProducts] = useState([]);
+
+    useEffect(() => {
+        (async () => {
+            try {
+                const { data } = await axios.get("/api/products");
```

Comment:
there is no need to call the API here as the only place where you will need to call the API is the products page.

Reply:
I did this because gonna need the products in two places the home page for the products section and in the product lisiting page aswell so thought this would be the better way around it .

Reply:
if you want to use the products at multiple place then take out that axios call and make a custom axios hook!